lib/jsdom/living/events/EventTarget-impl.js

```diff
@@ -0,0 +1,237 @@
+      targetParent = domSymbolTree.parent(targetParent);
+    }
+    if (event.type !== "load" && target._defaultView) { // https://html.spec.whatwg.org/#events-and-the-window-object
+      eventPath.push(target._defaultView);
+    }
+
+    eventImpl.eventPhase = Event.CAPTURING_PHASE;
+    for (let i = eventPath.length - 1; i >= 0; --i) {
+      if (eventImpl._stopPropagationFlag) {
+        break;
+      }
+
+      const object = eventPath[i];
+      const objectImpl = idlUtils.implForWrapper(object);
+      const eventListeners = objectImpl._eventListeners[event.type];
+      invokeEventListeners(eventListeners, object, event);
+    }
+
+    eventImpl.eventPhase = Event.AT_TARGET;
+    if (!eventImpl._stopPropagationFlag && this._eventListeners[event.type]) {
+      const eventListeners = this._eventListeners[event.type];
+      invokeEventListeners(eventListeners, eventImpl.target, event);
+    }
+
+    invokeInlineListeners(eventImpl.target, event);
+
+    if (event.bubbles) {
+      eventImpl.eventPhase = Event.BUBBLING_PHASE;
+      for (let i = 0; i < eventPath.length; ++i) {
+        if (eventImpl._stopPropagationFlag) {
+          break;
+        }
+
+        const object = eventPath[i];
+        const objectImpl = idlUtils.implForWrapper(object);
+        const eventListeners = objectImpl._eventListeners[event.type];
+        invokeEventListeners(eventListeners, object, event);
+
+        invokeInlineListeners(object, event);
+      }
+    }
+
+    eventImpl._dispatchFlag = false;
+    eventImpl.eventPhase = Event.NONE;
+    eventImpl.currentTarget = null;
+    return !eventImpl._canceledFlag;
+  }
+}
+
+module.exports = {
+  implementation: EventTargetImpl
+};
+
+function invokeInlineListeners(object, event) {
+  const inlineListener = getListenerForInlineEventHandler(object, event.type);
+  if (inlineListener) {
+    const document = object._ownerDocument || object._document;
+
+    // Will be falsy for windows that have closed
+    if (document) {
+      const implementation = document.implementation;
+
+      if (implementation._hasFeature("ProcessExternalResources", "script")) {
+        invokeEventListeners([{ callback: inlineListener }], object, event);
+      }
+    }
+  }
+}
+
+function invokeEventListeners(listeners, target, event) {
+  const eventImpl = idlUtils.implForWrapper(event);
+  eventImpl.currentTarget = target;
+  if (!listeners) {
+    return;
+  }
+
+  const handlers = listeners.slice();
+  for (let i = 0; i < handlers.length; ++i) {
+    if (eventImpl._stopImmediatePropagationFlag) {
+      return;
+    }
+
+    const listener = handlers[i];
+    if (listeners.indexOf(listener) === -1 ||
+        (event.eventPhase === Event.CAPTURING_PHASE && !listener.capture) ||
+        (event.eventPhase === Event.BUBBLING_PHASE && listener.capture)) {
+      continue;
+    }
+
+    try {
+      listener.callback.call(eventImpl.currentTarget, event);
+    } catch (e) {
+      let window = null;
+      if (target._document) {
```

Comment:
This is a bit weird - I had to rewrite it and check this a bit more explicitly, since there might be a Window which has its document removed, which logged a whole lot of errors to jsdomErrors.

I'm not sure why those errors weren't there before though.